org.jboss.tools.vscode.java/src/org/jboss/tools/vscode/java/internal/handlers/CompletionResolveHandler.java

```diff
@@ -43,8 +46,20 @@ CompletionItem resolve(CompletionItem param) {
 		// clean resolve data
 		param.setData(null);
 
-		if (data.containsKey(DATA_FIELD_URI) && data.containsKey(DATA_FIELD_DECLARATION_SIGNATURE)) {
-			ICompilationUnit unit = JDTUtils.resolveCompilationUnit(data.get(DATA_FIELD_URI));
+		if (!data.containsKey(DATA_FIELD_URI) || !data.containsKey(DATA_FIELD_REQUEST_ID) || !data.containsKey(DATA_FIELD_PROPOSAL_ID)) {
+			return param;
+		}
+		int proposalId = Integer.parseInt(data.get(DATA_FIELD_PROPOSAL_ID));
+		long requestId = Long.parseLong(data.get(DATA_FIELD_REQUEST_ID));
+		CompletionResponse completionResponse = CompletionResponses.get(requestId);
+		if (completionResponse == null || completionResponse.getProposals().size() <= proposalId) {
+			return param;
```

Comment:
This will send the client a  completionItem with no response perhaps we should just error in this case